sdk/go/common/resource/resource_id.go

```diff
@@ -128,3 +130,87 @@ func NewUniqueHexV2(urn URN, sequenceNumber int, prefix string, randlen, maxlen
 
 	return prefix + hex.EncodeToString(bs)[:randlen], nil
 }
+
+type cryptoSource struct{}
+
+func (c *cryptoSource) Int63() int64 {
+	return int64(c.Uint64() & ((1 << 63) - 1))
+}
+func (cryptoSource) Uint64() uint64 {
+	var data uint64
+	err := binary.Read(cryptorand.Reader, binary.LittleEndian, &data)
+	contract.AssertNoError(err)
+	return data
+}
+func (cryptoSource) Seed(seed int64) {
+	panic("We don't expect or support this method being called")
+}
+
+type hashSource struct {
+	hasher hash.Hash
+	seed   []byte
+}
+
+func (src *hashSource) Int63() int64 {
+	return int64(src.Uint64() & ((1 << 63) - 1))
+}
+func (src *hashSource) Uint64() uint64 {
+	// hashSource works by re-hashing the same seed over and over.
+	n, err := src.hasher.Write(src.seed)
+	contract.Assert(n == len(src.seed))
+	contract.AssertNoError(err)
+
+	randomBytes := src.hasher.Sum(nil)
+	// We expect hasher to be a SHA256 hash function
+	contract.Assert(len(randomBytes) == 32)
+
+	// Collapse the 32 bytes down to 8
+	seed :=
+		binary.LittleEndian.Uint64(randomBytes[0:]) ^
+			binary.LittleEndian.Uint64(randomBytes[8:]) ^
+			binary.LittleEndian.Uint64(randomBytes[16:]) ^
+			binary.LittleEndian.Uint64(randomBytes[24:])
+
+	return seed
+}
+func (hashSource) Seed(seed int64) {
+	panic("We don't expect or support this method being called")
+}
```

Comment:
`Seed` is an interface function on [`Source`](https://pkg.go.dev/math/rand#Source). We're not using it but need to implement it so that we count as a `Source` and can be passed to [`rand.New`](https://pkg.go.dev/math/rand#New).